src/task/TaskManager.ts

```diff
@@ -0,0 +1,71 @@
+import {TaskClient} from "./TaskClient"
+import os from "os"
+import {RunnerOptions, TaskRunner} from "./TaskRunner"
+import {ConductorLogger, DefaultLogger} from "../common/ConductorLogger"
+import {ConductorWorker} from "./Worker"
+
+export interface TaskManagerConfig {
+  logger?: ConductorLogger
+  options?: Partial<RunnerOptions>
+}
+
+const defaultRunnerOptions: Required<RunnerOptions> = {
+  workerID: '',
+  pollInterval: 1000,
+  maxRunner: 1,
+  domain: ''
+}
+
+function workerId (options: Partial<RunnerOptions>) {
+  return options.workerID ?? os.hostname()
+}
+
+export class TaskManager {
+  private tasks: Record<string, Array<TaskRunner>> = {}
+  private readonly client: TaskClient
+  private readonly logger: ConductorLogger
+  private workers: Array<ConductorWorker>
+  private runnerOptions: Required<RunnerOptions>
+
+  constructor(client: TaskClient, workers: Array<ConductorWorker>, config: TaskManagerConfig = {}) {
+    if (!workers) { throw new Error("No workers supplied to TaskManager. Please pass an array of workers.") }
+    this.client = client
+    this.logger = config.logger ?? new DefaultLogger()
+    this.workers = workers
+    const providedOptions = config.options ?? {}
+    this.runnerOptions = {
+      ...defaultRunnerOptions,
+      workerID: workerId(providedOptions),
+    }
+  }
+
+  startPolling = () => {
+    const maxRunner = this.runnerOptions.maxRunner ?? 1
+    this.workers.forEach(worker => {
+      this.tasks[worker.taskDefName] = []
+      for (let i = 0; i < maxRunner; i++) {
```

Comment:
I am the least confident in this part of the code. We could offer concurrency via `worker_threads` but we would gain very little since Node is perfectly suited for these sorts of I/O tasks. 

Instead, I choose to follow a similar model to the existing codebase (a promise loop) but move the knob for concurrency into the manager (starting multiple polling `TaskRunners`) instead of having 1 `TaskRunner` handle capacity.